api/graphql/schema.graphql

```diff
@@ -396,74 +404,76 @@ enum EducationStatus {
+    title: String
+    description: String
+    startDate: Date @rename(attribute: "start_date")
+    endDate: Date @rename(attribute: "end_date")
+    details: String
+    skills: [Skill!] @morphToMany
+    skillPivot: ExperienceSkillPivot @rename(attribute: "pivot")
 }
 type CommunityExperience implements Experience {
-  id: ID!
-  applicant: Applicant! @belongsTo(relation: "user")
-  title: String
-  organization: String
-  project: String
-  startDate: Date @rename(attribute: "start_date")
-  endDate: Date @rename(attribute: "end_date")
-  details: String
-  experienceSkills: [ExperienceSkill] @morphMany
+    id: ID!
+    applicant: Applicant! @belongsTo(relation: "user")
+    title: String
+    organization: String
+    project: String
+    startDate: Date @rename(attribute: "start_date")
+    endDate: Date @rename(attribute: "end_date")
+    details: String
+    skills: [Skill!] @morphToMany
+    skillPivot: ExperienceSkillPivot @rename(attribute: "pivot")
 }
 type EducationExperience implements Experience {
-  id: ID!
-  applicant: Applicant! @belongsTo(relation: "user")
-  institution: String
-  areaOfStudy: String @rename(attribute: "area_of_study")
-  thesisTitle: String @rename(attribute: "thesis_title")
-  startDate: Date @rename(attribute: "start_date")
-  endDate: Date @rename(attribute: "end_date")
-  type: EducationType
-  status: EducationStatus
-  details: String
-  experienceSkills: [ExperienceSkill] @morphMany
+    id: ID!
+    applicant: Applicant! @belongsTo(relation: "user")
+    institution: String
+    areaOfStudy: String @rename(attribute: "area_of_study")
+    thesisTitle: String @rename(attribute: "thesis_title")
+    startDate: Date @rename(attribute: "start_date")
+    endDate: Date @rename(attribute: "end_date")
+    type: EducationType
+    status: EducationStatus
+    details: String
+    skills: [Skill!] @morphToMany
+    skillPivot: ExperienceSkillPivot @rename(attribute: "pivot")
 }
 type AwardExperience implements Experience {
-  id: ID!
-  applicant: Applicant! @belongsTo(relation: "user")
-  title: String
-  issuedBy: String @rename(attribute: "issued_by")
-  awardedDate: Date @rename(attribute: "awarded_date")
-  awardedTo: AwardedTo @rename(attribute: "awarded_to")
-  awardedScope: AwardedScope @rename(attribute: "awarded_scope")
-  details: String
-  experienceSkills: [ExperienceSkill] @morphMany
+    id: ID!
+    applicant: Applicant! @belongsTo(relation: "user")
+    title: String
+    issuedBy: String @rename(attribute: "issued_by")
+    awardedDate: Date @rename(attribute: "awarded_date")
+    awardedTo: AwardedTo @rename(attribute: "awarded_to")
+    awardedScope: AwardedScope @rename(attribute: "awarded_scope")
+    details: String
+    skills: [Skill!] @morphToMany
+    skillPivot: ExperienceSkillPivot @rename(attribute: "pivot")
 }
 
-type ExperienceSkill {
-  id: ID!
-  details: String
-  skill: Skill! @belongsTo
-  experience: Experience! @morphTo
+type ExperienceSkillPivot {
```

Comment:
Are we able to get rid of this type?  Is it necessary? If we're trying to align with how the other ones are done, the pivot tables are invisible to the schema, I think.  The experience interface has its own `details` property now, right?

Reply:
I'm pretty sure we still need this since we're just querying through this to get the details property of the relationship. The details property on experiences is for details on the experience itself.

Reply:
Ah, OK, is it this?
![image](https://user-images.githubusercontent.com/8978655/161590565-6510c533-6f02-42b0-bd55-3169c6102fc8.png)
I dislike the name - it's leaking implementation details.  I'm having trouble coming up with a better one, though. :laughing: ExperienceSkillUsage?  ExperienceSkillRelationsip?  ExperienceSkillAssignment?

Reply:
Yeah, it's that. I can see your complaint about the name but I'm also unsure about the best replacement.

Reply:
Jerbo has recommended "[experience-skill-record](https://talent-cloud.slack.com/archives/CFCE2FSR4/p1649101910273059?thread_ts=1649094859.312789&cid=CFCE2FSR4)" and Tristan is on board.  So I suggest ExperienceSkillRecord for the schema type.  Hopefully that will be clearer when communicating with designers and users.  I noticed that the table name should be experience_skill (without the s) to comply with convention but I think that's out of scope.  I opened #2403 for that.